### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by matteomattei** [postcss/postcss-safe-parser#15]
Hello, trying parsing a CSS rule I got an error like the one reported in the subject.
This is a snippet to recreate the issue:

```
const postcss = require("postcss");
const safe = require("postcss-safe-parser");

const style = `.ReadMsgBody{ width:100%; } .ExternalClass{ width:100%; } html,body{ background-color:#ffffff; color:#9b042b; font-family:Verdana, Geneva, sans-serif; font-size:11px; text-align:left; -webkit-text-size-adjust:none !important; } h1{ color:#111111 !important; font-size:16px !important; ; line-height:16px !important; display:inline; font-weight:normal; margin-bottom:10px !important; } h2{ font-weight:bold; font-size:12px !important; display:inline; line-height:16px !important; font-family:verdana; } h3{ font-size:13px !important; font-weight:bold; margin:0px !important; padding:0px !important; padding-bottom:5px !important; color:#111111 !important; } h4{ font-size:12px; font-weight:bold !important; margin:0px !important; padding:0px !important; padding-bottom:5px !important; color:#111111 !important; } h5{ font-size:11px; font-weight:bold; margin:0px !important; padding:0px !important; padding-bottom:5px !important; color:#111111 !important; } h6{ font-size:11px; font-weight:bold; margin:0px !important; padding:0px !important; padding-bottom:5px !important; color:#111111 !important; } a,a:visited{ color:#020202; text-decoration:none !important; font-size:10px; } a:hover{ color:#ff0505 !important; } img{ display:block !important; border:0px !important; margin:0px !important; padding:0px !important; } p{ margin:0px; padding:0px; font-size:10px !important; line-height:13px !important; text-align:left; font-family:verdana; } .footer{ color:#FFFFFF !important; text-align:center !important; } .footer p{ text-align:center !important; } .footer a{ color:#FFFFFF !important; } .footer p a{ color:#FFFFFF !important; } .footer_links{ color:#FFFFFF !important; } sub,sup{ font-size:75%; line-height:0; position:relative; vertical-align:baseline; } sup{ top:-0.5em; } sub{ bottom:-0.25em; } .tableBorder{ border-width:1px; border-style:inset; border-color:gray; background-color:white; } .red_link{ font-weight:bold !important; font-size:11px !important; color:#ff0505 !important; text-decoration:none; !important;:; } @media print{ #_t { background-image: url('https://ly7fgofo.emltrk.com/ly7fgofo?p&d=[UNIQUE]');}} div.OutlookMessageHeader {background-image:url('https://ly7fgofo.emltrk.com/ly7fgofo?f&d=[UNIQUE]')} table.moz-email-headers-table {background-image:url('https://ly7fgofo.emltrk.com/ly7fgofo?f&d=[UNIQUE]')} blockquote #_t {background-image:url('https://ly7fgofo.emltrk.com/ly7fgofo?f&d=[UNIQUE]')} #MailContainerBody #_t {background-image:url('https://ly7fgofo.emltrk.com/ly7fgofo?f&d=[UNIQUE]')}`;

postcss()
  .process(style, { from: undefined, parser: safe })
  .then(function(result) {
    console.log(result.content);
  });
```

And this is the error I get:

```
(node:9679) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '0' of undefined
    at SafeParser.decl (/tmp/aaa/node_modules/postcss/lib/parser.js:203:21)
    at SafeParser.decl (/tmp/aaa/node_modules/postcss-safe-parser/lib/safe-parser.js:48:30)
    at SafeParser.other (/tmp/aaa/node_modules/postcss/lib/parser.js:133:18)
    at SafeParser.parse (/tmp/aaa/node_modules/postcss/lib/parser.js:77:16)
    at safeParse (/tmp/aaa/node_modules/postcss-safe-parser/lib/safe-parse.js:11:10)
    at new LazyResult (/tmp/aaa/node_modules/postcss/lib/lazy-result.js:60:16)
    at Processor.<anonymous> (/tmp/aaa/node_modules/postcss/lib/processor.js:138:12)
    at Processor.process (/tmp/aaa/node_modules/postcss/lib/processor.js:117:23)
    at Object.<anonymous> (/tmp/aaa/app.js:10:4)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
(node:9679) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:9679) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Any idea on how to fix it?
Thanks.

**Comment by ai** [comment]
Hi. Can you help me and reduce the size of `style`?

Any way I will be able to look at this issue only tomorrow or after tomorrow.

**Comment by matteomattei** [comment]
I don't know what rule is broken... anyway with this change seems working. So it might be worth enforcing some checks and eventually remove the breaking rule.

```
--- node_modules/postcss/lib/parser_old.js	1985-10-26 09:15:00.000000000 +0100
+++ node_modules/postcss/lib/parser.js	2020-02-07 10:15:25.668978552 +0100
@@ -200,15 +200,17 @@
       };
     }
 
-    while (tokens[0][0] !== 'word') {
+    while (tokens.length && tokens[0][0] !== 'word') {
       if (tokens.length === 1) this.unknownWord(tokens);
       node.raws.before += tokens.shift()[1];
     }
 
-    node.source.start = {
-      line: tokens[0][2],
-      column: tokens[0][3]
-    };
+    if(tokens.length){
+        node.source.start = {
+          line: tokens[0][2],
+          column: tokens[0][3]
+        };
+    }
     node.prop = '';
 
     while (tokens.length) {
```

**Comment by matteomattei** [comment]
I think I found the error:

```
.red_link {
	font-weight: bold !important;
	font-size: 11px !important;
	color: #ff0505 !important;
	text-decoration: none;
	!important;
	: ;
}
```

**Comment by ai** [comment]
Fixed d790e76

**Comment by ai** [comment]
I may need a few hours to update dependencies.

**Comment by ai** [comment]
Released at 4.0.2

<hr>

If this fix helped your business, talk with the manager for [funding PostCSS 8.0](https://opencollective.com/postcss/). Without community support, I soon will not be able to spend time on issues.

**Comment by matteomattei** [comment]
Thank you!!! It seems working great! ;-)
